test/unit/signup/account_manager_test.rb

```diff
@@ -126,6 +126,20 @@ class ProviderAccountManagerTest < Signup::AccountManagerTest
       assert_equal 'API', account.first_service!.name
     end
 
+    test 'creating a provider will create default Backend API if api_as_product is disabled' do
```

Comment:
really? :scream: shouldn't it be when it is enabled, it creates it, and when it is disabled, it does not?

Reply:
No that is correct
When it is on API as Product, we do not create a default backend API per service

Reply:
@hallelujah And when API as Product is disabled, we create a backend API for the service... what for? if api as product is disabled, we don't need a backend api at all, no?

Reply:
We need it because we moved the private endpoint to BackendApi
We create a default one with echo-api